xla/hlo/ir/hlo_module.cc

```diff
@@ -1243,30 +1243,30 @@ void CopyUniqueIds(const HloModule& source, HloModule* clone,
 
 }  // namespace
 
-std::unique_ptr<HloModule> HloModule::Clone(
-    const std::string& suffix,
-    std::optional<const HloModuleConfig> config_in) const {
-  auto module = CreateModule(suffix, config_in, *this);
-
-  HloCloneContext context(module.get(), suffix);
+void HloModule::Clone(const std::string& suffix, HloCloneContext* context,
+                      std::optional<const HloModuleConfig> config) const {
+  CHECK(context != nullptr) << "HloCloneContext pointer must not be null.";
```

Comment:
If we make it a private method, I am not sure whether we really need the CHECKs here.